capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module

```diff
@@ -190,3 +199,56 @@ function c4m_features_og_events_facet_items_alter(&$build, &$settings) {
     }
   }
 }
+
+/**
+ * Implements hook_entity_info_alter().
+ *
+ * Add "Upcoming event" view mode.
+ */
+function c4m_features_og_events_entity_info_alter(&$entity_info) {
+
+  $entity_info['node']['view modes']['upcoming_event'] = array(
+    'label' => t('Upcoming event'),
+    'custom settings' => TRUE,
+  );
+}
+
+/**
+ * Fetches upcoming events (Events in the future that haven't started yet),
+ * Depending on the user's role (guest, logged-in),
+ * This function fetches the events from user's groups or public groups.
+ * 1 Event for anonymous user, 3 for logged in.
+ *
+ * @return string
+ *  HTML of rendered upcoming events.
+ */
+function c4m_features_og_events_upcoming_events() {
+  global $user;
+  $date = date('Y-m-d H:i:s', time());
```

Comment:
I asked Peter about what 'the most upcoming events' are and he told me, that these are events that start not earlier than tomorrow.